src/Build/BackEnd/Shared/TargetResult.cs

```diff
@@ -44,17 +44,24 @@ public class TargetResult : ITargetResult, ITranslatable
         /// </summary>
         private CacheInfo _cacheInfo;
 
+        /// <summary>
+        /// The (possibly null) <see cref="BuildEventContext"/> from the original target build
+        /// </summary>
+        private BuildEventContext _originalBuildEventContext;
+
         /// <summary>
         /// Initializes the results with specified items and result.
         /// </summary>
         /// <param name="items">The items produced by the target.</param>
         /// <param name="result">The overall result for the target.</param>
-        internal TargetResult(TaskItem[] items, WorkUnitResult result)
+        /// <param name="originalBuildEventContext">The original build event context from when the target was first built, if available. Can be null.</param>
```

Comment:
Can you please better document when this is null and when it isn't? I assume it's null when this target first executed, and after that all subsequent executions should have an OBEC.

Reply:
Good idea, added a comment.